R/method-plot.R

```diff
@@ -1234,38 +1236,56 @@ plotEncDimSearch.OUTRIDER <- function(object){
     if(is(object, 'OutriderDataSet')){
         if(!'encDimTable' %in% colnames(metadata(object)) &
                 !is(metadata(object)$encDimTable, 'data.table')){
-            stop('Please run first the findEncodingDim before ',
+            stop('Please run the estimateBestQ function before ',
                     'plotting the results of it.')
         }
         dt <- metadata(object)$encDimTable
-        q <- getBestQ(object)
+        opt_q <- getBestQ(object)
     } else {
         dt <- object
         dt <- dt[, opt:=
                 encodingDimension[which.max(evaluationLoss)[1]], by=zScore]
-        q <- dt[opt == encodingDimension, opt]
+        opt_q <- dt[opt == encodingDimension, opt]
     }
+    
+    if ('oht' %in% colnames(dt)){
+      # Singular value plot
+      opt_data <- dt[oht==TRUE,]
+      
+      ggplot(dt, aes(q, singular_values)) + 
+        geom_line(color="blue") +
+        geom_vline(xintercept = opt_data$q, linetype = "dashed", color = "darkgray") +
+        geom_text(data=opt_data, aes(x = q+3, y = singular_values, 
+                                     label = q), color = "darkgray", size=4.5) +
+        xlab("Singular value rank (Estimated q)") +
+        ylab("Log(singular value)") + 
```

Comment:
Please change to ylab("Singular value") ## There is a difference between plotting a value in the log scale and log-transforming the value and then plotting it. This plot corresponds to the first case. A lab saying "log(value)" corresponds to the second case.